homeassistant/components/tedee/lock.py

```diff
@@ -0,0 +1,151 @@
+"""Tedee lock entities."""
+from dataclasses import dataclass
+from typing import Any
+
+from pytedee_async import TedeeClientException, TedeeLock, TedeeLockState
+
+from homeassistant.components.lock import (
+    LockEntity,
+    LockEntityDescription,
+    LockEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import CONF_UNLOCK_PULLS_LATCH, DOMAIN
+from .coordinator import TedeeApiCoordinator
+from .entity import TedeeEntity, TedeeEntityDescription
+
+
+@dataclass(frozen=True, kw_only=True)
+class TedeeLockEntityDescription(LockEntityDescription, TedeeEntityDescription):
+    """Describes Tedee lock entity."""
+
+
+ENTITIES: tuple[TedeeLockEntityDescription, ...] = (
+    TedeeLockEntityDescription(
+        unique_id_fn=lambda lock: f"{lock.lock_id}-lock",
+        key="lock",
+        translation_key="lock",
```

Comment:
The lock is the main feature of this device, consider setting the name to None so the entity name will follow the device name.